app/controllers/idt/api/v2/distributions_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Idt::Api::V2::DistributionsController < Idt::Api::V1::BaseController
+  protect_from_forgery with: :exception
+  before_action :verify_access
+
+  # rubocop:disable Metrics/MethodLength, Naming/AccessorMethodName
+  def get_distribution
+    # rubocop:enable Metrics/MethodLength, Naming/AccessorMethodName
+    distribution_id = params[:distribution_id]
+    # Checks if the distribution id is blank and if it exists with the database
+    if distribution_id.blank? || !valid_id?(distribution_id)
+      render_error(400, "Distribution Does Not Exist Or Id is blank", distribution_id)
+      return
+    end
+
+    begin
+      # Retrieves the distribution package from the PacMan API
+      distribution = PacManService.get_distribution_request(distribution_id)
+      response_code = distribution.code
+      if response_code != 200
+        fail StandardError
+      end
+      # Handles errors when making any requests both from Pacman and the DB
+    rescue StandardError
+      case response_code
+      when 404
+        pending_establishment(distribution_id, response_code)
+      else
+        response_code = 500
+        render_error(response_code, "Internal Server Error", distribution_id)
+      end
+      return
+    end
+
+    render json: format_response(distribution)
+  end
+
+  def pending_establishment(distribution_id, response_code)
+    render json: { id: distribution_id, response_status: "PENDING_ESTABLISHMENT" }, status: response_code
```

Comment:
```suggestion
    render json: { id: distribution_id, status: "PENDING_ESTABLISHMENT" }, status: 200
```

The reason for this suggestion is so that the key name is the same as what is returned whenever a distro is successfully located in Pacman. I also think that we can return a 200 here and not forward the 404 along, as it's not an error.. yet (it will be if we can never establish the distribution remotely).

Reply:
I suppose `response_code` wouldn't need to be a param any longer in this case.

Reply:
This m,akes sense